pkg/Cpanel/Security/Advisor/Assessors/Mysql.pm

```diff
@@ -109,6 +113,32 @@ sub _check_for_anonymous_users {
     return 1;
 }
 
+sub _check_for_public_bind_address {
+    my $self = shift;
+
+    my $mycnf        = Cpanel::MysqlUtils::MyCnf::Full::etc_my_cnf();
+    my $bind_address = $mycnf->{'mysqld'}->{'bind-address'};
+
+    my @deny_rules   = grep { /--dport 3306/ && /-j (DROP|REJECT)/ } split /\n/, Cpanel::SafeRun::Errors::saferunnoerror( '/sbin/iptables',  '--list-rules' );
+    my @deny_rules_6 = grep { /--dport 3306/ && /-j (DROP|REJECT)/ } split /\n/, Cpanel::SafeRun::Errors::saferunnoerror( '/sbin/ip6tables', '--list-rules' );
+
+    # This is a very basic check.
+    # We are looking to see if the bind address unset or set to anything other than localhost.
+    # We also check to see if there are any likely deny/reject rules on the SQL port.
+
+    if ( ( defined($bind_address) && Cpanel::IP::Loopback::is_loopback($bind_address) ) || @deny_rules || @deny_rules_6 ) {
```

Comment:
If a server has both IPv4 and IPv6 and they drop only one of them, then will we say the connection is protected when it isn't?